File: testdata/langgraph/simple_chain.py
"""Sequential agent — START → classify → respond → END.

Three-node straight chain. Used as the baseline LangGraph test fixture:
no conditional edges, no loops, every node deterministically reachable.

Expected Shingan analysis: zero findings (graph is healthy).
"""
from typing import TypedDict

from langgraph.graph import StateGraph, START, END


class State(TypedDict):
    """Conversation state passed between nodes."""

    user_input: str
    classification: str
    response: str


def classify(state: State) -> State:
    """Classify the user's intent (mocked for fixture purposes)."""
    return {**state, "classification": "greeting"}


def respond(state: State) -> State:
    """Produce the assistant's reply based on the classification."""
    return {**state, "response": f"Hello, {state['user_input']}!"}


builder = StateGraph(State)
builder.add_node("classify", classify)
builder.add_node("respond", respond)
builder.add_edge(START, "classify")
builder.add_edge("classify", "respond")
builder.add_edge("respond", END)

graph = builder.compile()

File: testdata/langgraphjs/simple_chain.ts
// Minimal linear LangGraph.js workflow: START -> a -> b -> END.
// Exercises addNode + addEdge + START/END sentinel handling.
import { StateGraph, START, END } from "@langchain/langgraph";

const graph = new StateGraph({ channels: {} })
  .addNode("a", callA)
  .addNode("b", callB)
  .addEdge(START, "a")
  .addEdge("a", "b")
  .addEdge("b", END);

function callA(state: any) {
  return state;
}

function callB(state: any) {
  return state;
}

export const app = graph.compile();
